from: https://github.com/tobiasjakobi/linux-odroid-public/commit/5d9267a91bb3b1ef00a5b93742ac6e2ca4d828e6

From 5d9267a91bb3b1ef00a5b93742ac6e2ca4d828e6 Mon Sep 17 00:00:00 2001
From: Marek Szyprowski <m.szyprowski@samsung.com>
Date: Thu, 28 Mar 2019 14:51:53 +0100
Subject: [PATCH] soc: samsung: Fix Exynos4412 ISP CMU reset in AFTR mode

When CPUidle enters C1 (AFTR) mode on Exynos4412, the operation of ISP
power domain fails. This can be observed by the following messages:

Power domain ISP disable failed

in the kernel log after using any of the ISP related device. This is
caused by enforcing low power state on the ISP CMU, which is a part of
the ISP power domain by the PMU driver during entering AFTR state.
This however fails, because the ISP power domain may be already turned
off by the Exynos power domain driver. To fix this disable forcing
low-power mode of the ISP domain from the PMU driver.

Fixes: bfce552d0b10 ("drivers: soc: Add support for Exynos PMU driver")
Signed-off-by: Marek Szyprowski <m.szyprowski@samsung.com>
---
 drivers/soc/samsung/exynos4-pmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/samsung/exynos4-pmu.c b/drivers/soc/samsung/exynos4-pmu.c
index a7cdbf1aac0c..62dcc68ada59 100644
--- a/drivers/soc/samsung/exynos4-pmu.c
+++ b/drivers/soc/samsung/exynos4-pmu.c
@@ -131,7 +131,7 @@ static const struct exynos_pmu_conf exynos4412_pmu_config[] = {
 	{ S5P_CMU_RESET_MFC_LOWPWR,		{ 0x1, 0x0, 0x0 } },
 	{ S5P_CMU_RESET_G3D_LOWPWR,		{ 0x1, 0x0, 0x0 } },
 	{ S5P_CMU_RESET_LCD0_LOWPWR,		{ 0x1, 0x0, 0x0 } },
-	{ S5P_CMU_RESET_ISP_LOWPWR,		{ 0x1, 0x0, 0x0 } },
+	{ S5P_CMU_RESET_ISP_LOWPWR,		{ 0x0, 0x0, 0x0 } },
 	{ S5P_CMU_RESET_MAUDIO_LOWPWR,		{ 0x1, 0x1, 0x0 } },
 	{ S5P_CMU_RESET_GPS_LOWPWR,		{ 0x1, 0x0, 0x0 } },
 	{ S5P_TOP_BUS_LOWPWR,			{ 0x3, 0x0, 0x0 } },
